src/backend/src/services/users.services.ts

```diff
@@ -0,0 +1,142 @@
+    const requestedUser = await prisma.user.findUnique({ where: { userId } });
+
+    if (!requestedUser) throw new NotFoundException('User', userId);
+
+    const settings = await prisma.user_Settings.upsert({
+      where: { userId },
+      update: {},
+      create: { userId }
+    });
+
+    if (!settings) throw new NotFoundException('User Settings', userId);
+
+    return settings;
+  }
+
+  static async updateUserSettings(userId: number, defaultTheme: any, slackId: string): Promise<User_Settings> {
+    const user = await prisma.user.findUnique({ where: { userId } });
+    if (!user) throw new NotFoundException('User', userId);
+
+    const updatedSettings = await prisma.user_Settings.upsert({
+      where: { userId },
+      update: { defaultTheme, slackId },
+      create: { userId, defaultTheme, slackId }
+    });
+
+    return updatedSettings;
+  }
+
+  static async logUserIn(idToken: string, header: string): Promise<AuthenticatedUser> {
+    const client = new OAuth2Client(process.env.REACT_APP_GOOGLE_AUTH_CLIENT_ID);
+    const ticket = await client.verifyIdToken({
+      idToken,
+      audience: process.env.REACT_APP_GOOGLE_AUTH_CLIENT_ID
+    });
+
+    const payload = ticket.getPayload();
+    if (!payload) throw new Error('Auth server response payload invalid');
+    const { sub: userId } = payload; // google user id
+    // check if user is already in the database via Google ID
+    let user = await prisma.user.findUnique({
+      where: { googleAuthId: userId },
+      ...authUserQueryArgs
+    });
+
+    // if not in database, create user in database
+    if (!user) {
+      const emailId = payload['email']!.includes('@husky.neu.edu') ? payload['email']!.split('@')[0] : null;
+      const createdUser = await prisma.user.create({
+        data: {
+          firstName: payload['given_name']!,
+          lastName: payload['family_name']!,
+          googleAuthId: userId,
+          email: payload['email']!,
+          emailId,
+          userSettings: { create: {} }
+        },
+        ...authUserQueryArgs
+      });
+      user = createdUser;
+    }
+
+    // register a login
+    await prisma.session.create({
+      data: {
+        userId: user.userId,
+        deviceInfo: header
+      }
+    });
+
+    return authenticatedUserTransformer(user);
+  }
+
+  static async logUserInDev(userId: number, header: string): Promise<AuthenticatedUser> {
+    const user = await prisma.user.findUnique({
+      where: { userId },
+      ...authUserQueryArgs
+    });
+
+    if (!user) throw new NotFoundException('User', userId);
+
+    // register a login
+    await prisma.session.create({
+      data: {
+        userId: user.userId,
+        deviceInfo: header
+      }
+    });
+
+    return authenticatedUserTransformer(user);
+  }
+
+  static async updateUserRole(targetUserId: number, userId: number, role: Role): Promise<User> {
+    const user = await prisma.user.findUnique({ where: { userId } });
```

Comment:
this is where the getCurrentUser should be passed in by the controller instead. this is the user doing the action so we need to get that through auth instead of just trusting whatever they pass in a post body. also you can then delete userId from the route body and stuff